BaragonService/src/main/java/com/hubspot/baragon/service/config/EdgeCacheConfiguration.java

```diff
@@ -0,0 +1,80 @@
+package com.hubspot.baragon.service.config;
+
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.hubspot.baragon.service.edgecache.cloudflare.EdgeCacheClass;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class EdgeCacheConfiguration {
+
+  @JsonProperty
+  @NotNull
+  private boolean enabled = false;
+
+  @JsonProperty
+  @NotNull
+  private EdgeCacheClass edgeCache = EdgeCacheClass.CLOUDFLARE;
+
+  // The following three values are used by the Cloudflare EdgeCache implementation.
+  // TODO: It's kind of a smell to have implementation-specific configs in the generic EdgeCacheConfiguration block.
+  // TODO: Is there a better way of doing this?
```

Comment:
I think these fields are generic enough that they would work for more cases. Still thinking on ways we could make it more generic/reusable. I'll comment if I come up with something

Reply:
Hm, I'm not familiar with how Dropwizard YAML -> Config Class magic works specifically, but can we just have a `Map<String, String> options` in the `EdgeCacheConfiguration` class, and then document the specific supported options for each `EdgeCache` implementation?